packages/admin/admin/src/EditDialog.stackTabsAndGrid.test.tsx

```diff
@@ -0,0 +1,196 @@
+                                                        onClick={() => editDialogApi.current?.openAddDialog()}
+                                                        variant="contained"
+                                                        color="primary"
+                                                    >
+                                                        Add product
+                                                    </Button>
+                                                ),
+                                            },
+                                        }}
+                                        disableVirtualization
+                                    />
+                                </RouterTab>
+                                <RouterTab label="Customers" path="/customers" forceRender={true}>
+                                    Customers Page
+                                </RouterTab>
+                            </RouterTabs>
+                        </StackPage>
+                        <StackPage name="productEdit" title="Product">
+                            {() => (
+                                <MainContent fullHeight disablePadding>
+                                    <SaveBoundary>
+                                        <StackToolbar>
+                                            <ToolbarBackButton />
+                                            <ToolbarAutomaticTitleItem />
+                                            <ToolbarFillSpace />
+                                            <ToolbarActions>
+                                                <SaveBoundarySaveButton />
+                                            </ToolbarActions>
+                                        </StackToolbar>
+                                        <div>Product Edit Page</div>
+                                    </SaveBoundary>
+                                </MainContent>
+                            )}
+                        </StackPage>
+                    </StackSwitch>
+                </Stack>
+                <AddProductDialog dialogApiRef={editDialogApi} />
+            </>
+        );
+    }
+
+    it("should open add dialog when clicking on Add product button in grid toolbar", async () => {
+        const rendered = render(
+            <MuiThemeProvider theme={createTheme()}>
+                <Router history={history}>
+                    <StackWithGridAndEditDialog />
+                </Router>
+            </MuiThemeProvider>,
+        );
+
+        expect(rendered.getByText("Add product")).toBeInTheDocument();
+        rendered.getByText("Add product").click();
+        expect(screen.getByText("Add a new product")).toBeInTheDocument();
+    });
+
+    it("should open edit stack page when clicking on edit button in grid", async () => {
+        const rendered = render(
+            <MuiThemeProvider theme={createTheme()}>
```

Comment:
is wrapping the MuiThemeProvider really needed? As far as i see, those providers got already wrapped in test-utils custom render